filament/backend/src/metal/MetalDriver.mm

```diff
@@ -917,8 +921,22 @@
                         dstRect.left >= 0 && dstRect.bottom >= 0,
             "Source and destination rects must be positive.");
 
-    // We always blit to/from the COLOR0 attachment.
-    MetalRenderTarget::Attachment srcColorAttachment = srcTarget->getColorAttachment(0);
+    size_t srcAttachIndex = 0;
+    if (any(buffers & TargetBufferFlags::COLOR0)) {
+        srcAttachIndex = 0;
+    }
+    if (any(buffers & TargetBufferFlags::COLOR1)) {
+        srcAttachIndex = 1;
+    }
+    if (any(buffers & TargetBufferFlags::COLOR2)) {
+        srcAttachIndex = 2;
+    }
+    if (any(buffers & TargetBufferFlags::COLOR3)) {
+        srcAttachIndex = 3;
+    }
+    MetalRenderTarget::Attachment srcColorAttachment = srcTarget->getColorAttachment(srcAttachIndex);
+
+    // We always blit to the COLOR0 attachment.
```

Comment:
This seems unrelated?

Reply:
I originally added this because I thought we needed it (we're bliting from COLOR1), though it looks like you create a new RT for the blit so it isn't strictly needed. But the way it was written before was incorrect, so probably best to leave it.